File: Latihan/app/src/main/java/com/example/latihan/MainActivity.java
package com.example.latihan;

import static android.content.ContentValues.TAG;

import androidx.appcompat.app.AppCompatActivity;
import androidx.swiperefreshlayout.widget.SwipeRefreshLayout;

import android.os.Bundle;
import android.os.Handler;
import android.widget.ListView;
import android.widget.TextView;

import com.android.volley.RequestQueue;
import com.android.volley.Response;
import com.android.volley.VolleyError;
import com.android.volley.VolleyLog;
import com.android.volley.toolbox.JsonArrayRequest;
import com.android.volley.toolbox.Volley;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {
    public static final String URL = "http://192.168.43.232/CRUDVolley/select.php";
    List<Data> itemlist = new ArrayList<Data>();
    SwipeRefreshLayout swipe;
    ListView list;
    Adapter adapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        swipe = (SwipeRefreshLayout) findViewById(R.id.swipe);
        list = (ListView) findViewById(R.id.list);

        adapter = new Adapter(MainActivity.this, itemlist);
        list.setAdapter(adapter);

        swipe.setOnRefreshListener((SwipeRefreshLayout.OnRefreshListener) this);

        swipe.post(new Runnable() {
                       @Override
                       public void run() {
                           swipe.setRefreshing(true);
                           itemlist.clear();
                           adapter.notifyDataSetChanged();
                           callVolley();
                       }
                   }
        );
    }

    public void onRefresh() {
        itemlist.clear();
        adapter.notifyDataSetChanged();
        callVolley();
    }

    private void callVolley() {
        itemlist.clear();
        adapter.notifyDataSetChanged();
        swipe.setRefreshing(true);
        refresh(1000);

        //membuat request JSON
        JsonArrayRequest jArr = new JsonArrayRequest(URL, new Response.Listener<JSONArray>() {
            @Override
            public void onResponse(JSONArray response) {
                //Parsing json
                for (int i = 0; i < response.length();
                     i++) {
                    try {
                        JSONObject obj = response.getJSONObject(i);

                            Data item = new Data();

                            item.setId(obj.getString("id"));
                            item.setTemperature(obj.getInt("data_temperature"));
                            item.setHumidity(obj.getInt("data_humidity"));
                            item.setTimestamp(obj.getString("timestamp"));

                            //menambah item ke array
                            itemlist.add(item);
                        } catch (JSONException e) {
                            e.printStackTrace();
                        }
                    }

                    //notifikasi adanya perubahan data pada adapter
                    adapter.notifyDataSetChanged();
                    swipe.setRefreshing(false);
                }
            }, new Response.ErrorListener() {
                @Override
                public void onErrorResponse(VolleyError error) {
                    VolleyLog.d(TAG, "Error: " + error.getMessage());
                    swipe.setRefreshing(false);
                }
            });

            // menambah request ke request queue
            RequestQueue mRequestQueue = Volley.newRequestQueue(getApplicationContext());
            mRequestQueue.add(jArr);

        }

        private void refresh(int milisecond) {

            final Handler handler = new Handler();

            final Runnable runnable = new Runnable() {
                @Override
                public void run() {
                    callVolley();
                }
            };

            handler.postDelayed(runnable, milisecond);

        }
    }

File: MyTestingApp/app/src/main/java/com/myfauzan/mytestingapp/MainActivity.kt
package com.myfauzan.mytestingapp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.view.View
import android.widget.Button
import android.widget.TextView
import java.lang.StringBuilder

class MainActivity : AppCompatActivity(), View.OnClickListener {

    private lateinit var btnSetValue: Button
    private lateinit var tvText: TextView

    private var names = ArrayList<String>()

    companion object {
        private const val TAG = "MainActivity"
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        tvText = findViewById(R.id.tv_text)
        btnSetValue = findViewById(R.id.btn_set_value)
        btnSetValue!!.setOnClickListener(this)

        names.add("Narenda Wicaksono")
        names.add("Kevin")
        names.add("Yoza")
    }

    override fun onClick(view: View) {
        if (view.id == R.id.btn_set_value){
            Log.d(TAG, names.toString())
            val name = StringBuilder()
            for (i in 0..3) {
                name.append(names[i]).append("\n")
            }
            tvText.text  = name.toString()
        }
    }
}